src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx

```diff
@@ -166,6 +166,15 @@ const EditSamplesConfirmationModal = ({
     changedMetadata,
   ]);
 
+  function resetMetadataFromCheckedSamples() {
+    const structuredMetadata: SampleIdToEditMetadataWebform = {};
+    checkedSamples.forEach((item) => {
+      structuredMetadata[item.privateId] = structureInitialMetadata(item);
+    });
+    setChangedMetadata(null);
+    setMetadata(structuredMetadata);
+  }
+
   useMemo(() => {
```

Comment:
This feels like it should be using `useEffect` instead of `useMemo`. We're not actually returning a memoized value with `useMemo`, so a standard side-effect causing `useEffect` seems right to me.

Reply:
Did you purposefully go back to using `useMemo` over `useEffect`? I see in the previous commit you went to `useEffect`, but then the next commit (this one I'm commenting on) it's back to `useMemo`. Can't tell if that was purposeful or an accident.

Reply:
it was purposeful, I wasn't sure if i wanted to change it over or not, i'll change it back